web-common/src/features/sources/modal/submitAddDataForm.ts

```diff
@@ -51,35 +56,83 @@ export async function submitAddSourceForm(
     formValues,
   );
 
-  // Make a new <source>.yaml file
-  const newSourceFilePath = getFileAPIPathFromNameAndType(
-    formValues.name as string,
-    EntityType.Table,
-  );
-  await runtimeServicePutFile(instanceId, {
-    path: newSourceFilePath,
-    blob: compileSourceYAML(rewrittenConnector, rewrittenFormValues),
-    create: true,
-    createOnly: false, // The modal might be opened from a YAML file with placeholder text, so the file might already exist
-  });
+  // Check if the connector was rewritten to DuckDB
+  const isRewrittenToDuckDb = rewrittenConnector.name === "duckdb";
 
-  // Create or update the `.env` file
-  await runtimeServicePutFile(instanceId, {
-    path: ".env",
-    blob: await updateDotEnvWithSecrets(
-      queryClient,
-      rewrittenConnector,
-      rewrittenFormValues,
-      "source",
-    ),
-    create: true,
-    createOnly: false,
-  });
+  if (isRewrittenToDuckDb) {
+    // Connectors that get rewritten to DuckDB (GCS, S3, Azure, etc.) create source files
+    const newSourceName = getName(
+      formValues.name as string,
+      fileArtifacts.getNamesForKind(ResourceKind.Source),
+    );
+
+    const newSourceFilePath = getFileAPIPathFromNameAndType(
+      newSourceName,
+      EntityType.Table,
+    );
+    await runtimeServicePutFile(instanceId, {
+      path: newSourceFilePath,
+      blob: compileSourceYAML(rewrittenConnector, rewrittenFormValues),
+      create: true,
+      createOnly: false,
+    });
+
+    // Create or update the `.env` file
+    await runtimeServicePutFile(instanceId, {
+      path: ".env",
+      blob: await updateDotEnvWithSecrets(
+        queryClient,
+        rewrittenConnector,
+        rewrittenFormValues,
+        "source",
+      ),
+      create: true,
+      createOnly: false,
+    });
+
+    await goto(`/files/${newSourceFilePath}`);
+  } else {
+    // Connectors that don't get rewritten (BigQuery, Athena, etc.) create connector files
+    const newConnectorName = getName(
+      connector.name as string,
+      fileArtifacts.getNamesForKind(ResourceKind.Connector),
+    );
 
-  await goto(`/files/${newSourceFilePath}`);
+    const newConnectorFilePath = getFileAPIPathFromNameAndType(
+      newConnectorName,
+      EntityType.Connector,
+    );
+    await runtimeServicePutFile(instanceId, {
+      path: newConnectorFilePath,
+      blob: compileConnectorYAML(connector, formValues),
+      create: true,
+      createOnly: false,
+    });
+
+    // Create or update the `.env` file
+    await runtimeServicePutFile(instanceId, {
+      path: ".env",
+      blob: await updateDotEnvWithSecrets(
+        queryClient,
+        connector,
+        formValues,
+        "connector",
+      ),
+      create: true,
+      createOnly: false,
+    });
+
+    await goto(`/files/${newConnectorFilePath}`);
+  }
```

Comment:
Why does this `connector` branch happen in the `submitAddSourceForm` function versus happen in the `submitAddConnectorForm` function? It's hard to understand which codepath the BigQuery connector would take, given BigQuery is mentioned in both places.

Can we move/remove this branch?